src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -1287,102 +1306,122 @@ bool emberAfThermostatClusterSetActivePresetRequestCallback(
     return true;
 }
 
-bool emberAfThermostatClusterStartPresetsSchedulesEditRequestCallback(
-    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
-    const chip::app::Clusters::Thermostat::Commands::StartPresetsSchedulesEditRequest::DecodableType & commandData)
+bool validAtomicAttributes(const chip::app::Clusters::Thermostat::Commands::AtomicRequest::DecodableType & commandData)
 {
-    ScopedNodeId sourceNodeId = GetSourceScopedNodeId(commandObj);
-
-    EndpointId endpoint = commandPath.mEndpointId;
-
-    // If the presets are editable and the scoped node id of the client sending StartPresetsSchedulesEditRequest command
-    // is not the same as the one that previously originated a StartPresetsSchedulesEditRequest command, return BUSY.
-    if (gThermostatAttrAccess.GetPresetsEditable(endpoint) &&
-        (gThermostatAttrAccess.GetOriginatorScopedNodeId(endpoint) != sourceNodeId))
+    auto attributeIdsIter = commandData.attributeRequests.begin();
+    bool requestedPresets, requestedSchedules;
```

Comment:
Don't those need to be initialized to false for the logic below to work right?  Right now we are going to get random behavior.